src/client/utils/tracking/gtm.ts

```diff
@@ -102,9 +104,15 @@ export const trackOfferGTM = (
   switchedFrom?: OfferData,
 ) => {
   const contractType = getContractType(offerData)
-  const initialOffer = getInitialOfferFromSessionStorage(contractType)
+  const contractCategory = getTrackableContractCategory(contractType)
   const grossPrice = Math.round(Number(offerData.cost.monthlyGross.amount))
   const netPrice = Math.round(Number(offerData.cost.monthlyNet.amount))
+
+  let initialOffer = getInitialOfferFromSessionStorage()
+  if (!initialOffer) {
+    setInitialOfferToSessionStorage(contractCategory)
```

Comment:
I think this is a perfect case for `useEffect` since it's causing a "side effect". Now we don't really know how many times this function will be called. Not a big deal but I think it would feel more react-y.

Reply:
Whoops, I didn't realise that from the diff :P nvm then!